transaction/transaction.go

```diff
@@ -73,6 +78,106 @@ func NewTxInput(hash []byte, index uint32) *TxInput {
 	}
 }
 
+type AddIssuanceArg struct {
+	Precision           uint
+	Contract            *IssuanceContract
+	AssetAmount         float64
+	TokenAmount         float64
+	AssetAddress        string
+	TokenAddress        string
+	ReissuanceTokenFlag uint
+	Net                 network.Network
+}
+
+//AddIssuance adds non-confidential issuance to the transaction
+func (tx *Transaction) AddIssuance(arg AddIssuanceArg) (uint32, error) {
+	if len(tx.Inputs) == 0 {
+		return 0, errors.New("transaction must contain at least one input ")
```

Comment:
return `-1` when returning error and remove space in error message:
```suggestion
		return 0, errors.New("transaction must contain at least one input")
```
This implies changing the return value from `uint32` to `int`

Reply:
Done